src/Libraries/PythonNodeModelsWpf/TabFoldingStrategy.cs

```diff
@@ -0,0 +1,164 @@
+using ICSharpCode.AvalonEdit.Document;
+using ICSharpCode.AvalonEdit.Folding;
+using System;
+using System.Collections.Generic;
+using System.Text;
+using System.Windows.Media.Animation;
+
+namespace PythonNodeModelsWpf
+{
+    /// <summary>
+    /// Allows producing tab based foldings
+    /// https://stackoverflow.com/questions/47224064/avalonedit-foldingstrategy-by-indent-python/47577500#47577500
+    /// </summary>
+    public class TabFoldingStrategy : AbstractFoldingStrategy
```

Comment:
internal?

Reply:
Yup will do that! This one is still very finicky, still working on making it more robust. It requires an update run virtually on each keystroke. I actually wanted to perhaps seek advice from you on how to profile performance with and without folding. Can I ask in Slack?